socorro/processor/mozilla_transform_rules.py

```diff
@@ -391,89 +391,23 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
         return True
 
 
-def setup_product_id_map(config, local_config, args_unused):
-    database_connection = local_config.database_class(local_config)
-    transaction = local_config.transaction_executor_class(
-        local_config,
-        database_connection
-    )
-    sql = (
-        "SELECT product_name, productid, rewrite FROM "
-        "product_productid_map WHERE rewrite IS TRUE"
-    )
-    product_mappings = transaction(
-        execute_query_fetchall,
-        sql
-    )
-    product_id_map = {}
-    for product_name, productid, rewrite in product_mappings:
-        product_id_map[productid] = {
-            'product_name': product_name,
-            'rewrite': rewrite
-        }
-    return product_id_map
-
-
 class ProductRewrite(Rule):
-    required_config = Namespace()
-    required_config.add_option(
-        'database_class',
-        doc="the class of the database",
-        default='socorro.external.postgresql.connection_context.'
-                'ConnectionContext',
-        from_string_converter=str_to_python_object,
-        reference_value_from='resource.postgresql',
-    )
-    required_config.add_option(
-        'transaction_executor_class',
-        default="socorro.database.transaction_executor."
-                "TransactionExecutorWithInfiniteBackoff",
-        doc='a class that will manage transactions',
-        from_string_converter=str_to_python_object,
-        reference_value_from='resource.postgresql',
-    )
-
-    required_config.add_aggregation(
-        'product_id_map',
-        setup_product_id_map
-    )
-
     def __init__(self, config):
         super(ProductRewrite, self).__init__(config)
-        self.product_id_map = setup_product_id_map(
-            config,
-            config,
-            None
-        )
+        self.product_map = {
+            "{aa3c5121-dab2-40e2-81ca-7ea25febc110}": "FennecAndroid",
+            "{99bceaaa-e3c6-48c1-b981-ef9b46b67d60}": "MetroFirefox",
+            "{webapprt@mozilla.org}": "WebappRuntime",
```

Comment:
Huh. I thought I pulled the values from production when I populated the static data for the local dev environment:

https://github.com/mozilla-services/socorro/blob/master/socorro/external/postgresql/staticdata.py#L233

Figured I'd mention it now even though I haven't double-checked it. I can double-check it tomorrow morning.

Reply:
I was hasty in writing this patch.

The `product_productid_map` table is used in a bunch of ADI-related tasks. The 'rewrite' field is used in this processor rule. Like the API endpoint, this is only considering rows where rewrite was true. The rest of the table should persist as long as ADI does, and it doesn't seem worth migrating out the rewrite column.

The only still-active product in that list is FennecAndroid, which is also the only `rewrite = t` field in the static version of the table. I think that this patch has unnecessary additional rows in the map, and your staticdata covers all of our actual use cases.

I'll amend this momentarily.